File: day03/day03.py
forest = list(open("day03_input.txt", "r").read().split("\n"))
height = len(forest)
width = len(forest[0])

# part 1
x = 3
y = 1
count = 0

while (y < height):
    if (x >= width):
        x = x - width
    if forest[y][x] == "#":
        count += 1
    x += 3
    y += 1
print(count)

# part 2
'''
memo = [["-" for i in range(width)] for i in range(height)]

def read(x, y):
    return memo[y][x]

def write(x, y, value):
    memo[y][x] = value
'''
def slope(x, y, count, step_x, step_y):
    if (y >= height):
        return count
    if (x >= width):
        x = x - width
    if forest[y][x] == "#":
        count += 1
    return slope(x + step_x, y + step_y, count, step_x, step_y)

a = slope(0,0,0,1,1)
b = slope(0,0,0,3,1)
c = slope(0,0,0,5,1)
d = slope(0,0,0,7,1)
e = slope(0,0,0,1,2)
print(a*b*c*d*e)

File: day03/day03.js
var fs = require('fs')
const { listenerCount } = require('process')
let dataset = fs.readFileSync('day03_input.txt').toString().split("\n")
let width = dataset[0].length
let height = dataset.length

function count(a) {
    dx = a[0]
    dy = a[1]
    total = 0
    x = 0
    y = 0
    while (y < height) {
        if (dataset[y][x] == '#') {
            total = total + 1
        }
        else { }
        x = (x + dx) % width
        y = y + dy
    }
    return total
}

let numberlist = [[1,1], [3,1], [5,1], [7,1], [1,2]]
console.log(count([3,1]))
console.log(numberlist.map(x => count(x)).reduce((a,b) => a * b))
